Mathlib/Data/ZMod/Basic.lean

```diff
@@ -821,6 +868,10 @@ theorem val_eq_zero : ∀ {n : ℕ} (a : ZMod n), a.val = 0 ↔ a = 0
     exact Iff.rfl
 #align zmod.val_eq_zero ZMod.val_eq_zero
 
+theorem val_ne_zero {n : ℕ}  (a : ZMod n) : a.val ≠ 0 ↔ a ≠ 0 := by
+  rw [not_iff_not]
+  apply val_eq_zero
```

Comment:
```suggestion
theorem val_ne_zero {n : ℕ} (a : ZMod n) : a.val ≠ 0 ↔ a ≠ 0 :=
  val_eq_zero.not
```
or similar

Reply:
Hm, maybe we shouldn't have that lemma at all, then...

Reply:
I think we generally are in favor of these `ne_zero` lemmas, even though the proof is trivial.